v2/slogger/rolling_file_appender/error.go

```diff
@@ -0,0 +1,157 @@
+package rolling_file_appender
+
+import (
+	"fmt"
+)
+
+type CloseError struct {
```

Comment:
It's currently being used during rotation: https://github.com/mongodb/slogger/blob/time-based-rotate/v2/slogger/rolling_file_appender/rolling_file_appender.go#L319 .  I will change Close() to use it as well